ids/node_id.go

```diff
@@ -60,35 +52,60 @@ func (id *NodeID) UnmarshalJSON(b []byte) error {
 	}
 
 	var err error
-	*id, err = NodeIDFromString(str[1:lastIndex])
+	*nodeID, err = NodeIDFromString(str[1:lastIndex])
 	return err
 }
 
-func (id *NodeID) UnmarshalText(text []byte) error {
-	return id.UnmarshalJSON(text)
+func (nodeID NodeID) MarshalText() ([]byte, error) {
+	return []byte(nodeID.String()), nil
+}
+
+func (nodeID *NodeID) UnmarshalText(text []byte) error {
+	return nodeID.UnmarshalJSON(text)
 }
 
-func (id NodeID) Less(other NodeID) bool {
-	return bytes.Compare(id[:], other[:]) == -1
+func (nodeID NodeID) Less(other NodeID) bool {
+	return nodeID < other
 }
 
-// ToNodeID attempt to convert a byte slice into a node id
-func ToNodeID(bytes []byte) (NodeID, error) {
-	nodeID, err := ToShortID(bytes)
-	return NodeID(nodeID), err
+func (nodeID NodeID) Equal(other NodeID) bool {
+	return nodeID == other
+}
+
+func NodeIDFromBytes(bytes []byte) NodeID {
+	return NodeID(string(bytes))
+}
+
+// NodeIDFromShortNodeID attempt to convert a byte slice into a node id
+func NodeIDFromShortNodeID(nodeID ShortNodeID) NodeID {
+	return NodeID(string(nodeID.Bytes()))
 }
 
 func NodeIDFromCert(cert *staking.Certificate) NodeID {
-	return hashing.ComputeHash160Array(
+	bytes := hashing.ComputeHash160Array(
 		hashing.ComputeHash256(cert.Raw),
 	)
+	return NodeID(string(bytes[:]))
+}
+
+func (nodeID NodeID) ToSize(newSize int) NodeID {
```

Comment:
Instead of having a `ToSize` to dynamically change the size of the id I feel like it would be cleaner to just enforce a constraint when creating this like so; I generally try to avoid mutable state where it's possible:
```
NewNodeID(s string) (NodeID, error) {
    if len(s) > NodeIDLen {
        return "", errors.New("invalid node id")
    }

    return "NodeID-" + s
}